packages/sdk-js/src/GrowthBook.ts

```diff
@@ -174,7 +177,14 @@ export class GrowthBook<
   }
 
   public setUserId(userId: string): void {
+    if (this._ctx.remoteEval) {
+      unsubscribe(this);
+    }
     this._ctx.userId = userId;
+    this._refresh({ skipCache: this.isRemoteEval() }, false, false);
```

Comment:
Same as setting attributes, we don't want to call this unless loadFeatures has already been called.

Reply:
good catch